### DISCUSSION TITLE
Usage of Lyrics metadata

### DISCUSSION THREAD

**Discussion by landgenoot** [librespot-org/librespot#1186] [General]
There seems to be [some work](https://github.com/librespot-org/librespot/commit/f72048e5e1a6f74c5476eef89a1de37fa7bb118f) done on fetching Lyrics from the Spotify API.
However, it is not clear to me how to use this functionality when running Librespot.

The lyrics are [not broadcasted along with the events](https://github.com/librespot-org/librespot/blob/dev/src/player_event_handler.rs). And as far as I understand, there is no API to communicate with the Librespot service. Is that correct?

How am I supposed to fetch the Lyrics?

**Comment by kingosticks** [comment]
Librespot is a library you can use in your own program, you can write code to fetch lyrics using that.

There's also a binary "example" program called librespot, this doesn't have any lyrics support added yet but someone could add that.
